---
title: "Jornada_16S"
author: "Nat Pombubpa"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.


### STEP1: Load all necessary packages for analysis using Phyloseq

More information about Phyloseq can be found at the following link:
[Phyloseq](https://joey711.github.io/phyloseq/)

If you get error in this step, you probably need to install any packages which causes error.

```{r warning=FALSE, message=FALSE}
library(phyloseq)
library(microbiomeSeq)
library(ape)
library(vegan)
library(reshape2)
library(ggplot2)
library(igraph)
library(gridExtra)
library(grid)
library(plyr)
```

### STEP2: Import Mapping file (metadata file)

1.Check mapping file before import to R, R doesn't seem to like sample name to start with number or contain "-" in sample name. If you get error in this step, you should check file name first.

2.First column of first row should not start with #, R will not read the first row that starts with #

```{r}
meta = read.table("050517NP515F-mapping2_NP_1_30.txt",header=TRUE,row.names=1,sep="\t",stringsAsFactors=FALSE)
```

### STEP3: Check if your metadata file has been import sucessfully and correctly

The output will show a table of your metadata file (mapping file).

*If you do not have header, you might start your first row with #

```{r}
head(meta)
```

### STEP4: Construct sample_data-class using imported metadata 

```{r}
sampleData <- sample_data(meta)
```



### STEP5: Import otu table

Otu table from Jornada 16S data is "Sam1_34a.otu_table.txt".

```{r}
otus <- read.table("Sam1_34a.otu_table.txt",header=T,sep="\t",row.names=1)
otumat <- as(as.matrix(otus), "matrix")
OTU = otu_table(otumat, taxa_are_rows = TRUE)
```

Check imported OTU table 

```{r}
head(OTU)
```



### STEP6: Import taxonomy table

Taxonmy table generated from AMPtk need to be rearranged using following script.

"perl rdp_taxonmy2mat.pl<Input_taxonmy.txt>Output_taxonomy.txt"

rdp_taxonomy2mat.pl was created by Professor Jason E. Stajich

```{r}
taxmat <- read.table("Sam1_34a.taxonomy.fix.txt", header=T,sep="\t",row.names=1)
taxmat <- as(as.matrix(taxmat),"matrix")
TAX = tax_table(taxmat)
```

### STEP7: Import phylogenetic tree

Phylogenetic tree can also be include for furthur phylogenetic analysis.

```{r warning=FALSE}
treefile = "Sam1_34a.tree.phy"
tree = read.tree(treefile)
```

### STEP8: Construct Phyloseq object

To construct phyloseq object, otu table, taxonomy table, and sampleData are required.
Phylogenetic tree can be included, but it is not necessary for constructing phyloseq object.

```{r warning=FALSE, message=FALSE}
ps = phyloseq(OTU,TAX,sampleData,tree)
```

Check phyloseq object

```{r}
ps
```


### STEP9: Remove singletons

Remove any OTUs that present only one time.

```{r message=FALSE}
ps.prune = prune_taxa(taxa_sums(ps) > 1, ps)
```

Rarefy otu table to even depth.
(This step was not necessary according to the phyloseq author.)

```{r warning=FALSE, message=FALSE}
ps.prune.rarefy = rarefy_even_depth(ps.prune, trimOTUs = FALSE)
```


### STEP10: Plot Alpha Diversity using Plot_Richness command

Alpha diversity measure: Shannon (VegZone)

```{r message=FALSE}
plot_richness(ps.prune.rarefy, x="VegZone", color=("VegZone"), measures=c( "Shannon")) + geom_boxplot() + ggtitle("Alpha Diversity (Shannon) by Veggitation Zone")
```

Alpha diversity measure: Chao1 (VegZone)

```{r}
plot_richness(ps.prune.rarefy, x="VegZone", color=("VegZone"), measures=c( "Chao1")) + geom_boxplot() + ggtitle("Alpha Diversity (Chao1) by Veggitation Zone")
```

### Reorder Alpha Diversity X-axis and legend 

Use "scale_x_discrete"" to reorder x-axis and "scale_color_discrete"" to reorder legend

```{r}

plot_richness(ps.prune.rarefy, x="VegZone", color=("VegZone"), measures=c( "Chao1")) + geom_boxplot() + ggtitle("Alpha Diversity (Chao1) by Veggitation Zone") + scale_x_discrete(limits=c("Mesquite","Grassland", "Creosote", "Tarbush", "Playa")) + scale_color_discrete(breaks=c("Mesquite","Grassland", "Creosote", "Tarbush", "Playa"))

```


```{r}
plot_richness(ps.prune.rarefy, x="VegZone", color=("VegZone"), measures=c( "Shannon")) + geom_boxplot() + ggtitle("Alpha Diversity (Shannon) by Veggitation Zone") + scale_x_discrete(limits=c("Mesquite","Grassland", "Creosote", "Tarbush", "Playa")) + scale_color_discrete(breaks=c("Mesquite","Grassland", "Creosote", "Tarbush", "Playa"))
```


Alpha diversity measure: Shannon (Disturbance)

```{r warning=FALSE}
plot_richness(ps.prune.rarefy, x="Disturbance", color=("Disturbance"), measures=c( "Chao1","Shannon", "Observed")) + geom_boxplot() + ggtitle("Alpha Diversity by Disturbance")
```

### STEP11: Calculate distance

```{r}
ps.ord <- ordinate(ps.prune, "PCoA", "unifrac")
```

### STEP12: Plot ordination

```{r}
psVegZone = plot_ordination(ps.prune, ps.ord, type = "samples", color = "VegZone") + stat_ellipse(geom = "polygon", alpha = 1/18, aes(fill = VegZone)) + theme_bw() + ggtitle("Beta Diversity (PCoA) by VegZone")
```

### STEP13: Print Plot

```{r}
print(psVegZone)
```

```{r}
psDisturbance = plot_ordination(ps.prune, ps.ord, type = "samples", color = "Disturbance") + stat_ellipse(geom = "polygon", alpha = 1/18, aes(fill = Disturbance)) + theme_bw() + ggtitle("Beta Diversity (PCoA) by Disturbance")
```

```{r}
print(psDisturbance)
```

